lib/evmone/eof.cpp

```diff
@@ -395,26 +395,35 @@ std::variant<EOFValidationError, int32_t> validate_max_stack_height(
 
             if (code_types[fid].outputs == NON_RETURNING_FUNCITON)
             {
-                stack_height_required = static_cast<int8_t>(code_types[fid].inputs);
+                stack_height_required = code_types[fid].inputs;
             }
             else
             {
                 if (code_types[func_index].outputs < code_types[fid].outputs)
                     return EOFValidationError::jumpf_destination_incompatible_outputs;
 
-                stack_height_required =
-                    static_cast<int8_t>(code_types[func_index].outputs + code_types[fid].inputs -
-                                        code_types[fid].outputs);
+                stack_height_required = code_types[func_index].outputs + code_types[fid].inputs -
+                                        code_types[fid].outputs;
                 if (stack_heights[i] > stack_height_required)
                     return EOFValidationError::stack_higher_than_outputs_required;
             }
         }
         else if (opcode == OP_RETF)
         {
-            stack_height_required = static_cast<int8_t>(code_types[func_index].outputs);
+            stack_height_required = code_types[func_index].outputs;
             if (stack_height > code_types[func_index].outputs)
                 return EOFValidationError::stack_higher_than_outputs_required;
         }
+        else if (opcode == OP_DUPN)
+        {
+            const auto n = static_cast<uint8_t>(code[i + 1]);
+            stack_height_required = n + 1;
+        }
+        else if (opcode == OP_SWAPN)
+        {
+            const auto n = static_cast<uint8_t>(code[i + 1]);
+            stack_height_required = n + 2;
```

Comment:
Ah, yes. This is because `n` is promoted to `int` in `n + 2`. So this is correct just the `static_cast` is unnecessary. How about `stack_height_required = code[i + 1] + 2`?

Reply:
yup, this works, I thought I started with it and it didn't work :shrug:, maybe because of that `int8_t` up there. Now it works. Thanks for bearing with me here.